bundler/lib/bundler/dsl.rb

```diff
@@ -447,6 +447,15 @@ def check_path_source_safety
     end
 
     def check_rubygems_source_safety
+      unless @sources.implicit_global_source?
```

Comment:
It should be `if @sources.implicit_global_source?`, but the logic of the method is also reversed, so it works like this too thanks to the double negation :sweat_smile:.